src/Elastic.Transport/Components/TransportClient/HttpRequestInvoker.cs

```diff
@@ -165,6 +169,10 @@ private async ValueTask<TResponse> RequestCoreAsync<TResponse>(bool isAsync, Req
 				response = requestData.ConnectionSettings.ProductRegistration.ResponseBuilder.ToResponse<TResponse>
 						(requestData, ex, statusCode, responseHeaders, responseStream, mimeType, contentLength, threadPoolStats, tcpStats);
 
+			// Defer disposal of the response message
+			if (response is StreamResponse sr)
```

Comment:
Ugly but i think its better to pass `receive` (as IDisposable) to `ResponseBuilder.ToResponse`. To ensure we inject to to `StreamResponse`'s constructor. 

If an exception happens in `ToResponse()` we might never set the finalizer.